app/server/controllers/extend-deadline.ts

```diff
@@ -0,0 +1,50 @@
+import { Router, Request, Response, NextFunction } from 'express';
+import * as AppInsights from 'app/server/app-insights';
+import * as Paths from 'app/server/paths';
+
+function getIndex(req: Request, res: Response) {
+  return res.render('extend-deadline/index.html', {});
+}
+
+function extensionConfirmation(getAllQuestionsService: any, extendDeadlineService: any) {
+
+  return async (req: Request, res: Response, next: NextFunction) => {
+
+    const hearingId: string = req.session.hearing.online_hearing_id;
+    const extend: string = req.body['extend-deadline'];
+
+    if (!extend) return res.render('extend-deadline/index.html', { error: true });
+
+    try {
+
+      let deadline: string;
+
+      if (extend === 'yes') {
+        const response = await extendDeadlineService(hearingId);
+        deadline = response.deadline_expiry_date;
+      } else {
+        const response = await getAllQuestionsService(hearingId);
```

Comment:
We could potentially think about storing the deadline date on the session to prevent the need to make this API call. Probably one for a separate PR.

Reply:
Ya, had considered this, decided to do it this way for simplicity, but happy to move to session

Reply:
Happy to keep it like this for now and move it to the session separately if that's easier?

Reply:
Have converted to session